### PR TITLE
fix(scripts): Code quality issues

### PR DESCRIPTION
**Pull request by wizard-28**
## Purpose

The scripts have many code quality issues.

## Approach

Fix them.

## Progress

- [x] `add-repo.sh`
- [x] `download.sh`
- [x] `error_log.sh`
- [x] `install-local.sh`
- [x] `query-info.sh` (nothing to do)
- [x] `remove.sh`
- [x] `search.sh`
- [x] `update.sh`
- [x] `upgrade.sh`

### CODE REVIEW

File: misc/scripts/install-local.sh [hunk 1]
```diff
@@ -483,24 +472,23 @@ if [[ $REMOVE_DEPENDS = y ]]; then
...
 
 fancy_message info "Symlinking files"
-cd /usr/src/pacstall/ || sudo mkdir -p /usr/src/pacstall && cd /usr/src/pacstall
+cd /usr/src/pacstall/ || sudo mkdir -p /usr/src/pacstall && cd /usr/src/pacstall || exit 1
 # By default (I think), stow symlinks to the directory behind it (..), but we want to symlink to /, or in other words, symlink files from pkg/usr to /usr
 if ! command -v stow > /dev/null; then
 	# If stow failed to install, install it
 	sudo apt-get install stow -y
-	cd /usr/src/pacstall
+	cd /usr/src/pacstall || exit 1
```
Comment (Henryws) [inline misc/scripts/install-local.sh:486]: `{ fancy_message error "/usr/src/pacstall does not exist" && exit 1 }`

Comment (wizard-28) [reply misc/scripts/install-local.sh]: Fixed in https://github.com/pacstall/pacstall/pull/276/commits/d7e37217811b08651b372e9e47cb6ce6b6357858

File: misc/scripts/install-local.sh [hunk 2]
```diff
@@ -141,16 +141,16 @@ function makeVirtualDeb {
 	# implements $(gives) variable
 	fancy_message info "Creating dummy package"
 	sudo mkdir -p "$SRCDIR/$name-pacstall/DEBIAN"
-	printf "Package: $name\n" | sudo tee "$SRCDIR/$name-pacstall/DEBIAN/control" > /dev/null
+	printf "Package: %s\n" "$name" | sudo tee "$SRCDIR/$name-pacstall/DEBIAN/control" > /dev/null
```
Comment (Henryws) [inline misc/scripts/install-local.sh:144]: what does this do?

File: misc/scripts/install-local.sh [hunk 3]
```diff
@@ -372,14 +365,14 @@ function hashcheck() {
 
 fancy_message info "Retrieving packages"
 mkdir -p "$SRCDIR"
-cd "$SRCDIR"
+cd "$SRCDIR" 2> /dev/null || ( fancy_message warn "Couldn't enter ${SRCDIR}" && exit 1)
```
Comment (Henryws) [inline misc/scripts/install-local.sh:368]: no smasher thingys. It should be `could not`

File: misc/scripts/install-local.sh [hunk 4]
```diff
@@ -372,14 +365,14 @@ function hashcheck() {
...
 
 case "$url" in
 	*.git)
 		# git clone quietly, with no history, and if submodules are there, download with 10 jobs
 		sudo git clone --quiet --depth=1 --jobs=10 "$url"
 		# cd into the directory
-		cd ./*/ 2> /dev/null
+		cd ./*/ 2> /dev/null || ( fancy_message warn "Couldn't enter into the cloned git repository" && exit 1)
```
Comment (Henryws) [inline misc/scripts/install-local.sh:375]: no smasher thingys. It should be `could not`

File: misc/scripts/install-local.sh [hunk 5]
```diff
@@ -390,32 +383,29 @@ case "$url" in
...
-		hashcheck "${url##*/}"
-		if [[ $? -ne 0 ]]; then
+		if ! hashcheck "${url##*/}"; then
 			return 1
 		fi
 		# unzip file
 		sudo unzip -q "${url##*/}" 1>&1 2>/dev/null
 		# cd into it
-		cd ./*/ 2> /dev/null
+		cd ./*/ 2> /dev/null || ( fancy_message warn "Couldn't enter into the downloaded archive" && exit 1)
```
Comment (Henryws) [inline misc/scripts/install-local.sh:392]: no smasher thingys. It should be `could not`

File: misc/scripts/install-local.sh (hunk 2 above)
Comment (wizard-28) [reply misc/scripts/install-local.sh:144]: https://github.com/koalaman/shellcheck/wiki/SC2182

File: misc/scripts/install-local.sh (hunk 3 above)
Comment (wizard-28) [reply misc/scripts/install-local.sh:368]: They are called contractions

File: misc/scripts/install-local.sh (hunk 4 above)
Comment (wizard-28) [reply misc/scripts/install-local.sh:375]: They are called contractions. Refactoring now.

File: misc/scripts/install-local.sh (hunk 5 above)
Comment (wizard-28) [reply misc/scripts/install-local.sh:392]: They are called contractions. Refactoring now.

Comment (wizard-28) [reply misc/scripts/install-local.sh]: Fixed in https://github.com/pacstall/pacstall/pull/276/commits/0691fb5e7635a168f4f677bc139b3ec1a9983765

File: misc/scripts/install-local.sh (hunk 4 above)
Comment (wizard-28) [reply misc/scripts/install-local.sh]: Fixed in https://github.com/pacstall/pacstall/pull/276/commits/0691fb5e7635a168f4f677bc139b3ec1a9983765

File: misc/scripts/install-local.sh (hunk 3 above)
Comment (wizard-28) [reply misc/scripts/install-local.sh]: Fixed in https://github.com/pacstall/pacstall/pull/276/commits/0691fb5e7635a168f4f677bc139b3ec1a9983765

File: misc/scripts/install-local.sh (hunk 5 above)
Comment (Henryws) [reply misc/scripts/install-local.sh]: > They are called contractions. Refactoring now.

Yeah, I guess...

File: misc/scripts/install-local.sh [hunk 6]
```diff
@@ -372,14 +365,14 @@ function hashcheck() {
 
 fancy_message info "Retrieving packages"
 mkdir -p "$SRCDIR"
-cd "$SRCDIR"
+cd "$SRCDIR" 2> /dev/null || ( fancy_message warn "Could not enter ${SRCDIR}" && exit 1)
```
Comment (D-Brox) [inline misc/scripts/install-local.sh:368]: Add error logging

Comment (D-Brox) [reply misc/scripts/install-local.sh:368]: Same for the others

File: misc/scripts/remove.sh [hunk 7]
```diff
@@ -56,7 +52,8 @@ case "$url" in
 	;;
 
 	*)
-		cd "$STOWDIR" || (sudo mkdir -p "$STOWDIR"; cd "$STOWDIR")
+		sudo mkdir -p "$STOWDIR"
+		cd "$STOWDIR" 2> /dev/null || ( fancy_message error "Could not enter ${STOWDIR}" && exit 1 )
```
Comment (D-Brox) [inline misc/scripts/remove.sh:56]: Add error logging

**Review by D-Brox — CHANGES_REQUESTED** [review]

File: misc/scripts/install-local.sh (hunk 6 above)
Comment (wizard-28) [reply misc/scripts/install-local.sh:368]: Can you suggest the correct snippet once? Then I can adapt the snippet for the others afterwards.

Comment (D-Brox) [reply misc/scripts/install-local.sh:368]: Something like `error_log 1 "install $PACKAGE"`.

Comment (D-Brox) [reply misc/scripts/install-local.sh:368]: Something like `error_log 1 "install $PACKAGE"`.

Comment (wizard-28) [reply misc/scripts/install-local.sh:368]: okay, I'll add it in soon.

File: misc/scripts/remove.sh (hunk 7 above)
Comment (wizard-28) [reply misc/scripts/remove.sh:56]: okay, I'll add it in soon.

Comment (wizard-28) [reply misc/scripts/remove.sh]: Fixed in https://github.com/pacstall/pacstall/pull/276/commits/99e0e97dd340f72b3f73673958e5ead2e5370b67

File: misc/scripts/install-local.sh (hunk 6 above)
Comment (wizard-28) [reply misc/scripts/install-local.sh]: Fixed in https://github.com/pacstall/pacstall/pull/276/commits/99e0e97dd340f72b3f73673958e5ead2e5370b67

File: misc/scripts/upgrade.sh [hunk 8]
```diff
@@ -23,24 +23,25 @@
 # along with Pacstall. If not, see <https://www.gnu.org/licenses/>.
 
 function ver_compare() {
-	local first="$(echo ${1} | sed 's/^[^0-9]*//')"
-	local second="$(echo ${2} | sed 's/^[^0-9]*//')"
+	local first second
+	first="$(echo "${1}" | sed 's/^[^0-9]*//')"
```
Comment (D-Brox) [inline misc/scripts/upgrade.sh:27]:
```suggestion
	first="${1#${1/[0-9]*/}}"
```

File: misc/scripts/upgrade.sh [hunk 9]
```diff
@@ -23,24 +23,25 @@
...
+	second="$(echo "${2}" | sed 's/^[^0-9]*//')"
```
Comment (D-Brox) [inline misc/scripts/upgrade.sh:28]:
```suggestion
	second="${2#${2/[0-9]*/}}"
```

**Review by D-Brox — CHANGES_REQUESTED** [review]

File: misc/scripts/remove.sh [hunk 10]
```diff
@@ -56,7 +52,8 @@ case "$url" in
 	;;
 
 	*)
-		cd "$STOWDIR" || (sudo mkdir -p "$STOWDIR"; cd "$STOWDIR")
+		sudo mkdir -p "$STOWDIR"
+		cd "$STOWDIR" 2> /dev/null || ( error_log 1 "install $PACKAGE"; fancy_message error "Could not enter ${STOWDIR}"; exit 1 )
```
Comment (D-Brox) [inline misc/scripts/remove.sh:56]:
```suggestion
		cd "$STOWDIR" 2> /dev/null || ( error_log 1 "remove $PACKAGE"; fancy_message error "Could not enter ${STOWDIR}"; exit 1 )
```

**Review by D-Brox — CHANGES_REQUESTED** [review]

**Review by D-Brox — APPROVED** [review]

**Review by Henryws — APPROVED** [review]
Noice

**Review by NarutoXY — APPROVED** [review]

**Pull request merged**